src/Storages/System/StorageSystemUsers.cpp

```diff
@@ -119,6 +119,7 @@ void StorageSystemUsers::fillData(MutableColumns & res_columns, ContextPtr conte
     auto & column_grantees_except_offsets = assert_cast<ColumnArray &>(*res_columns[column_index++]).getOffsets();
     auto & column_default_database = assert_cast<ColumnString &>(*res_columns[column_index++]);
 
+    // todo arthur check this
```

Comment:
I forgot to look into this one, gimme a few

Reply:
Hm.. this one might be a bit trickier

Reply:
@pufit do you have an opinion on how this should evolve? If I understand correctly, this is about the users table. Should we turn the `auth_type` column into a list and add more json objects on `auth_data`?

Reply:
I converted the `auth_type` column from `Int8` to `Array(Int8)` and kept `auth_params` as a `String`, but populated it with a JSON Array instead of a single JSON object.

It might make more sense a be more idiomatic to make it an array as well..

Reply:
Changed it to an array of strings=

Reply:
> I forgot to look into this one, gimme a few

Yes, I missed it during the review as well (kinda awkward 😅)

> It might make more sense a be more idiomatic to make it an array as well..

Yes, I think `Array(Int8)` and `Array(String)` should work

Reply:
Can you also add a test on system.users with multiple auth methods?

Reply:
Fixed existing settings and added a new one